assertj-core/src/main/java/org/assertj/core/error/ShouldBeMarked.java

```diff
@@ -23,11 +24,35 @@ public static ErrorMessageFactory shouldBeMarked(AtomicMarkableReference<?> actu
     return new ShouldBeMarked(actual, true);
   }
 
+  /**
+   * Creates a new <code>{@link ShouldBeMarked}</code>.
+   *
+   * @param actual the actual buffer in the failed assertion.
+   * @return the created {@code ErrorMessageFactory}.
+   */
+  public static ErrorMessageFactory shouldBeMarked(Buffer actual) {
```

Comment:
I don't know whether mixing Errors from AtomicMarkableReference and ByteBuffer is a good idea - they are separate abstractions even if they both have a "marker".
@joel-costigliola - is there any strict convention for re-using ErrorMessageFactories inside AsssertJ?